backend/functions/src/v2/course/csv/functions.ts

```diff
@@ -0,0 +1,64 @@
+import { SurveyQuestion } from "../../../common/types";
+import { storage } from "../../../common/db";
+import { Request, Response } from "express";
+import { getCourseRefByDocId } from "../../../common/utils";
+import { FieldInfo, parseAsync } from "json2csv";
+import { v4 as uuidv4 } from "uuid";
+import * as fs from "fs";
+import * as path from "path";
+import * as os from "os";
+
+export const postCourseCSV = async (req: Request, res: Response) => {
+	const courseId = req.params.courseId;
+	// eslint-disable-next-line @typescript-eslint/no-unused-vars
+	const courseDocRef = (await getCourseRefByDocId(courseId).catch((err) => {
+		return res.status(404).send(`Course with id ${courseId} is not found`);
+	})) as FirebaseFirestore.DocumentReference;
+
+	// fetch headers for csv file
+	const courseDocSnapshot = await courseDocRef.get();
+	const fields: (
+		| string
+		| FieldInfo<FirebaseFirestore.DocumentData>
+	)[] = (courseDocSnapshot.get("question") as SurveyQuestion[]).map((q) => {
+		const questionHash = q.question.hash;
+		return {
+			label: questionHash,
+			value: "surveyResponse.".concat(questionHash),
+		};
+	});
+	fields.push("studentId", "fullName");
```

Comment:
should "studentId" be replaced with "email"? At this point I will leave it to @XiaochenSeanYu and you to decide which one is more appropriate.

Reply:
i already refactored to email; would be inconvenient to change back + i don't see why not since email is a more generalized ID if this were to expand beyond cornell

Reply:
alright then then change all to "email" it is

Reply:
okay then change to "email" it is.